scripts/net/run-sample-tests.sh

```diff
@@ -0,0 +1,408 @@
+
+image=net-tools
+name=net-tools
+network=net-tools0
+zephyr_pid=0
+docker_pid=0
+configuration=""
+result=0
+sample=""
+
+check_dirs ()
+{
+    local ret_zephyr=0
+    local ret_net_tools=0
+
+    if [ -z "$ZEPHYR_BASE" ]
+    then
+	echo '$ZEPHYR_BASE is unset' >&2
+	ret_zephyr=1
+    fi
+
+    if [ ! -d "$ZEPHYR_BASE" ]
+    then
+	echo '$ZEPHYR_BASE is set, but it is not a directory' >&2
+	ret_zephyr=1
+    fi
+
+    if [ -z "$NET_TOOLS_BASE" ]
+    then
+	local d
+
+	for d in "$ZEPHYR_BASE/../.." "$ZEPHYR_BASE/.."
+	do
+	    local l
+
+	    l="$d/net-tools"
+	    if [ -d "$l" ]
+	    then
+		NET_TOOLS_BASE="$l"
+		break
+	    fi
+	done
+    fi
+
+    if [ $ret_zephyr -eq 0 ]
+    then
+	echo "\$ZEPHYR_BASE $ZEPHYR_BASE"
+    fi
+
+    if [ -z "$NET_TOOLS_BASE" ]
+    then
+	echo '$NET_TOOLS_BASE is unset, no net-tools found' >&2
+	ret_net_tools=1
+    fi
+
+    if [ ! -d "$NET_TOOLS_BASE" ]
+    then
+	echo '$NET_TOOLS_BASE set, but it is not a directory' >&2
+	ret_net_tools=1
+    fi
+
+    if [ $ret_net_tools -eq 0 ]
+    then
+	echo "\$NET_TOOLS_BASE $NET_TOOLS_BASE"
+    fi
+
+    if [ $ret_zephyr -ne 0 -o $ret_net_tools -ne 0 ]
+    then
+	return 1
+    fi
+
+    return 0
+}
+
+start_configuration ()
+{
+    local bridge_interface=""
+
+    if ! docker image ls | grep "$image" > /dev/null
+    then
+	echo "Docker image '$image' not found" >&2
+	return 1
+    fi
+
+    if ! docker network ls | grep "$network" > /dev/null
+    then
+	bridge_interface=$("$NET_TOOLS_BASE/net-setup.sh" \
+			   --config "$NET_TOOLS_BASE/docker.conf" \
+			   start 2>/dev/null | tail -1)
+	if [ $? != 0 ]
+	then
+	    echo "Could not start Docker network '$network'" >&2
+	    return 1
+	fi
+
+	echo "Started Docker network '$network' bridge interface" \
+	     "'$bridge_interface'..."
+    fi
+
+    if ! docker ps | grep "$name" > /dev/null
+    then
+	if ! docker run --hostname=$name --name=$name \
+             --ip=192.0.2.2 --ip6=2001:db8::2 \
+             --rm -dit --network=$network $image > /dev/null
+	then
+	    echo "Could not start Docker container '$image'"
+	    return 1
+	fi
+
+	echo "Started Docker container '$name'.."
+    fi
+
+}
+
+stop_configuration ()
+{
+    local bridge_interface=""
+
+    if docker ps | grep "$name" > /dev/null
+    then
+	docker stop "$name" > /dev/null
+	if [ $? -eq 0 ]
+	then
+	    echo "Stopped Docker container '$name'..."
+	else
+	    echo "Could not stop Docker container '$name'" >&2
+	fi
+    fi
+
+    bridge_interface=$(docker network ls | grep "$network" | cut -d " " -f 1)
+    if [ -n "$bridge_interface" ]
+    then
+	docker network rm "$network" > /dev/null
+	if [ $? -eq 0 ]
+	then
+	    echo "Stopped Docker network '$network' bridge interface" \
+		 "'br-$bridge_interface'..."
+	else
+	    echo "Could not stop Docker network '$network'" >&2
+	fi
+    fi
+}
+
+start_zephyr ()
+{
+    rm -rf build && mkdir build && \
+	cmake -GNinja -DBOARD=native_posix -B build && \
+	ninja -C build
```

Comment:
We could override the IP addresses here when compiling if needed (related to my earlier comment of echo-client prj.conf)

Reply:
Yes.